Bug/dss11 sc 127914 eks apiversion incompatibility with kubectl

jules-casoli opened the pull request:
kubectl deprecated the "alpha" apiVersion in version 1.24 
This selectively update the kube_config file to use the new apiVersion, depending on the installed versions of kubectl and aws-iam-authenticator (which needs to be > 0.5.4 to support the "beta" apiVersion)

amandineslx reviewed (changes requested):
* deploy DSS design node in EKS via FM (I used 11.4)
* create managed cluster via DSS
  * use default (latest = 1.25) Kubernetes version
* connect to the VM
* upgrade `kubectl` version (see [here](https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/))
```
sudo su root
cd /usr/local/bin
curl -LO https://dl.k8s.io/release/v1.26.0/bin/linux/amd64/kubectl
# validate you are using the right kubectl
which kubectl
```
* compress manually the plugin repo content
* update the plugin version in DSS to use the one from the PR
* go to the cluster page and go to tab `Actions`
* trigger the macro `Check and correct internal config`
  * :white_check_mark: there is a warning about the versions of `kubectl` and `aws-iam-authenticator` being misaligned
  * :warning: the warning is not very visible. Should we add a warning sign or some color?
* try to start the cluster
  * :warning: error is raised and is not very explicit
  ```
  [2023-03-28 09:02:52,290] [19849/MainThread] [INFO] [root] Create autoscaler with : ["kubectl", "get", "pods", "--namespace", "kube-system", "-l", "app=cluster-autoscaler", "--ignore-not-found"]
  Traceback (most recent call last):
    File "/opt/dataiku-dss-11.4.0/python/dataiku/cluster/server.py", line 50, in serve
      result = cluster.start()
    File "/tmp/tmp_folder_rJlaxzaP/dku_code.py", line 235, in start
    File "/data/dataiku/dss_data/plugins/installed/eks-clusters/python-lib/dku_kube/autoscaler.py", line 13, in add_autoscaler_if_needed
      if not has_autoscaler(kube_config_path):
    File "/data/dataiku/dss_data/plugins/installed/eks-clusters/python-lib/dku_kube/autoscaler.py", line 9, in has_autoscaler
      out, err = run_with_timeout(cmd, env=env, timeout=5)
    File "/data/dataiku/dss_data/plugins/installed/eks-clusters/python-lib/dku_kube/kubectl_command.py", line 29, in run_with_timeout
      raise KubeCommandException("Command failed with %s" % rv, out, err, rv)
  KubeCommandException: Command failed with 1
  [2023-03-28 09:02:53,297] [19849/MainThread] [INFO] [dataiku.base.socket_block_link] Client closed
  ```
* try to download the starting logs of the cluster
  * :warning: I am unable to unzip the archive
  ```
  Unable to expand "dtart_2023_03_28.log.gz" into "Downloads".
  (Error 79 - Inappropriate file type or format.)
  ```
* upgrade `aws-iam-authenticator` (see [here](https://docs.aws.amazon.com/eks/latest/userguide/install-aws-iam-authenticator.html))
```
sudo su root
cd /usr/local/bin
curl -Lo aws-iam-authenticator https://github.com/kubernetes-sigs/aws-iam-authenticator/releases/download/v0.5.9/aws-iam-authenticator_0.5.9_linux_amd64
chmod +x ./aws-iam-authenticator
# validate you are using the right aws-iam-authenticator
which aws-iam-authenticator
# if needed, delete the current one
```
* trigger the macro `Check and correct internal config`
  * :white_check_mark: everything fine in the checks now
* stop cluster
  * :white_check_mark: it works
* start cluster again
* check Kubeconfig
  * :white_check_mark: API version is `apiVersion: client.authentication.k8s.io/v1beta1`
```
sudo su dataiku
cd /data/dataiku/dss_data/clusters/asouilleux-cluster/exec
vi kube_config
```
* checking the cluster information in the UI
  * :warning: there is a remaining tag in the cluster data: `"eksctl.cluster.k8s.io/v1alpha1/cluster-name": "asouilleux-cluster"`

jules-casoli commented:
> the warning is not very visible. Should we add a warning sign or some color?

Indeed, changed the class to error:
<img width="936" alt="image" src="https://user-images.githubusercontent.com/91876294/228487529-e40b37c7-669b-472d-9d0f-910e9e3118c6.png">

> error is raised and is not very explicit

Added a check on start/attach. Not pretty, but good enough ?
<img width="590" alt="image" src="https://user-images.githubusercontent.com/91876294/228487682-1fbbe9d8-58c3-42a0-8206-877a7d1980ae.png">

>  ⚠️ there is a remaining tag in the cluster data: "eksctl.cluster.k8s.io/v1alpha1/cluster-name": "asouilleux-cluster"
    ⚠️ there are additional tags mentioning the alpha too

Actually, I started a cluster on 1.26/0.5.9, and they are still there. We don't have any `alpha` in our code, so I guess these come from eksctl, so, probably not an issue

marchelleboid commented on python-lib/dku_utils/tools_version.py:
```diff
@@ -0,0 +1,22 @@
+def check_versions():
+    kubectl_version = get_kubectl_version()
+    authenticator_version = get_authenticator_version()
+    if kubectl_should_use_beta_apiVersion(kubectl_version) and authenticator_version < '0.5.4':
```
Hopefully aws-iam-authenticator never gets to `0.10.x` 😰

marchelleboid reviewed (approved):
Looks good to me in general, ended up testing similar cases as Amandine. Errors seem improved now. I also tested attaching to clusters and checking that `kube_config` was created with a `v1beta1`, and also using the macro while the cluster was already running.

amandineslx reviewed (approved):
OK for me

marchelleboid reviewed (approved):
Didn't re-test, but macro removal looks good to me
